root/brent_test.go

```diff
@@ -0,0 +1,405 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package root_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"gonum.org/v1/exp/root"
+	"gonum.org/v1/gonum/floats/scalar"
+)
+
+var eps = math.Nextafter(1, 2) - 1
+
+func TestBrent(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		name string
+		f    func(float64) float64
+		a    float64
+		b    float64
+		tol  float64
+		want float64
+	}
+	var tests []testcase
+	for _, f := range tstutilsFns {
+		tests = append(tests, testcase{name: f.name, f: f.f, a: .5, b: math.Sqrt(3), tol: 4 * eps, want: 1})
+	}
+	for _, tc := range aps {
+		tests = append(tests, testcase{name: tc.name, f: tc.f, a: tc.a, b: tc.b, tol: 4 * eps, want: tc.root})
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s", test.name), func(t *testing.T) {
+			got, err := root.Brent(test.f, test.a, test.b, test.tol)
+			if !scalar.EqualWithinAbsOrRel(got, test.want, 2e-12, test.tol) {
+				if err != nil {
+					t.Fatalf("%s: %s", test.name, err)
+				}
+				t.Fatalf("%s: got %f want %f", test.name, got, test.want)
+			}
+		})
+	}
+}
+
+func TestScipyIssue5557(t *testing.T) {
+	t.Parallel()
+
+	f := func(x float64) float64 {
+		if x < 0.5 {
+			return -0.1
+		}
+		return x - 0.6
+	}
+	tol := 4 * eps
+	got, err := root.Brent(f, 0., 1., tol)
+	if err != nil {
+		t.Fatalf("Brent(%f, %f, %f): %s", 0., 1., tol, err)
+	}
+	const want = 0.6
+	if !scalar.EqualWithinRel(got, want, tol) {
+		t.Fatalf("Brent(%f, %f, %f): got %f want %f", 0., 1., tol, got, want)
```

Comment:
```suggestion
	got, err := root.Brent(f, 0, 1, tol)
	if err != nil {
		t.Fatalf("Brent(%f, %f, %f): %s", 0, 1, tol, err)
	}
	const want = 0.6
	if !scalar.EqualWithinRel(got, want, tol) {
		t.Fatalf("Brent(%f, %f, %f): got %f want %f", 0, 1, tol, got, want)
```
(untyped constants in a typed slot don't need extra syntax)

Reply:
Thanks for the suggestion, I have applied them to the function call to `Brent`.
However, for the call to `Fatalf`, string formatting for %f seems to insist on the additional dot, thus I have kept it.

Reply:
Good point; the reason for that is that the args are `any`, so the default type is `int` when there is no syntax to say otherwise. Please add the following zero then; `, 0.0, 1.0`.

Reply:
Yes, I have applied this change in a hopefully more principled way, by defining the parameters as float64 variables, and inserting them throughout.